chroma_api/stratagem.py

```diff
@@ -0,0 +1,145 @@
+from django.db import models
+from django.db.models import Q
+from django.core.exceptions import ObjectDoesNotExist
+from toolz.functoolz import pipe
+
+import tastypie.http as http
+
+from tastypie import fields
+from tastypie.resources import Resource
+from chroma_api.authentication import AnonymousAuthentication
+from chroma_core.services.job_scheduler.job_scheduler_client import JobSchedulerClient
+from tastypie.authorization import DjangoAuthorization
+from tastypie.validation import Validation
+from chroma_api.validation_utils import validate
+from chroma_api.utils import custom_response, dehydrate_command
+from chroma_core.models import (
+    StratagemConfiguration,
+    ManagedHost,
+    ManagedMdt,
+    ManagedTargetMount,
+    ManagedFilesystem,
+    Command,
+)
+
+from chroma_api.chroma_model_resource import ChromaModelResource
+
+
+class RunStratagemValidation(Validation):
+    def is_valid(self, bundle, request=None):
+        purge_duration = bundle.data.get("purge_duration")
+        report_duration = bundle.data.get("report_duration")
+
+        if "filesystem" not in bundle.data:
+            return {"code": "filesystem_required", "message": "filesystem required when running stratagem."}
+        elif isinstance(purge_duration, int) and isinstance(report_duration, int) and report_duration >= purge_duration:
```

Comment:
What happens if either of these are not ints? is that being validated? Perhaps cast both to ints (if they exist) and return an error if they are not.

Reply:
I attempt to convert these to ints and if it fails I return an error now.